capstone/src/main/java/com/google/sps/servlets/PostDataServlet.java

```diff
@@ -0,0 +1,93 @@
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import com.google.sps.Objects.User;
+import com.google.sps.Objects.Post;
+import com.google.sps.Objects.Comment;
+
+@WebServlet("/post")
+public class PostDataServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Query query = new Query("Post").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Post> posts = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      
+      long timestamp = (long) entity.getProperty("timestamp");
+      String authorId = (String) entity.getProperty("authorId");
+      String postText = (String) entity.getProperty("postText");
+      String challengeName = (String) entity.getProperty("challengeName");
+      String img = (String) entity.getProperty("img");
+      ArrayList<String> likes = (ArrayList<String>) entity.getProperty("likes");
+      ArrayList<Comment> comments = (ArrayList<Comment>) entity.getProperty("comments");
+      Post userPost = new Post(authorId, postText, comments, challengeName, timestamp, img, likes);
+      posts.add(userPost);
+    }
+
+    // Convert to json
+    response.setContentType("application/json;");
+    response.getWriter().println(new Gson().toJson(posts));
+  }
+
+
+   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Receives submitted post 
+    long timestamp = System.currentTimeMillis();
+    String authorId = "Jane Doe";
+    String postText = request.getParameter("post-input");
+    String challengeName = "Challenge Name";
+    String img = "";
+    ArrayList<String> likes = new ArrayList<>();
```

Comment:
I think likes should be the id of the users who have liked the post right? In that case, I think datastore returns a long for id